core/server/worker/src/main/java/alluxio/worker/block/annotator/BlockAnnotator.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.block.annotator;
+
+import alluxio.annotation.PublicApi;
+import alluxio.collections.Pair;
+import alluxio.conf.PropertyKey;
+import alluxio.conf.ServerConfiguration;
+import alluxio.util.CommonUtils;
+
+import javax.annotation.concurrent.ThreadSafe;
+import java.util.List;
+
+/**
+ * Interface for providers that annotates blocks for sorting.
```

Comment:
Could you clarify the responsibilities of this class? What is the division of work between this class and `DefaultBlockIterator`?